concordium-node/src/grpc2.rs

```diff
@@ -2615,3 +2646,115 @@ where
         })
     }
 }
+
+/// Connection stream with an attached semaphore and gauge.
+pub struct ConnStreamWithTicket {
+    stream:                 TcpIncoming,
+    /// The semaphore is used to keep track of the number of connections we
+    /// currently maintain. For each new connection we give out a ticket.
+    semaphore:              tokio_util::sync::PollSemaphore,
+    /// A gauge to record the number of connected client. This is incremented
+    /// when we accept a new connection and decremented on the [`drop`] of
+    /// the connection.
+    grpc_connected_clients: GenericGauge<AtomicU64>,
+}
+
+pub struct AddrStreamWithTicket {
+    addr:    AddrStream,
+    /// The permit is attached to the connection so that when the connection is
+    /// dropped the permit is also dropped, releasing a connection token.
+    /// Thus the permit itself is not used directly by the service, it is only
+    /// used for its drop behaviour.
+    #[allow(dead_code)]
+    permit:  OwnedSemaphorePermit,
+    counter: GenericGauge<AtomicU64>,
```

Comment:
```suggestion
    counter: IntGauge,
```

Reply:
These are not the same. Since the value is always going to be non-negative I prefer to use an unsigned value.